src/snowflake/cli/plugins/nativeapp/utils.py

```diff
@@ -82,3 +82,22 @@ def verify_no_directories(paths_to_sync: List[Path]):
 def verify_exists(path: Path):
     if not path.exists():
         raise ClickException(f"The following path does not exist: {path}")
+
+
+def get_all_file_paths_under_dir(directory: Path):
+    """
+    Recursively traverse the given directory and yield file paths. If the directory is a symlink, then the function follows the behavior of os.walk for symlinks.
+    The directory may be a relative path to the project root.
```

Comment:
Calling out that this relies on cwd == project root. Might want to add to the docstring.